src/app/workflow/view/view.component.ts

```diff
@@ -89,6 +91,50 @@ export class ViewWorkflowComponent extends View implements OnInit {
     });
   }
 
+  /**
+   * Updates the workflow version and alerts the Dockstore User with success
+   * or failure.
+   *
+   * @private
+   * @memberof ViewWorkflowComponent
+   */
+  private updateWorkflowToSnapshot(version): void {
+    this.workflowsService.updateWorkflowVersion(this.workflow.id, [version]).subscribe(
+      workflowVersions => {
+        this.alertService.detailedSuccess('Snapshot successfully created!');
+        const workflow = { ...this.workflowQuery.getActive() };
+        workflow.workflowVersions = workflowVersions;
+        this.workflowService.setWorkflow(workflow);
+      },
+      error => {
+        if (error) {
+          this.alertService.detailedError(error);
+        } else {
+          this.alertService.simpleError();
+        }
+      }
+    );
+  }
+  /**
+   * Opens a confirmation dialog that the Dockstore User can use to
+   * confirm they want a snapshot.
+   *
+   * @private
+   * @memberof ViewWorkflowComponent
+   */
+  snapshotVersion(): void {
+    // Create a new version temporarily with the frozen bit set. We assume the version is
+    // not already a snapshot since the UI controls shouldn't be available for
+    // snapshotted versions.
+    const snapshot = { ...this.version };
+    snapshot.frozen = true;
+    const confirmMessage = 'Snapshotting a version cannot be undone. Are you sure you want to snapshot this version?';
```

Comment:
Though it will likely never happen, double-check that the version is not already frozen before proceeding.  Also, make an issue to hook up the new confirmation dialog with this